### DISCUSSION TITLE
If our code does not follow the document but pass all the tests, are we treated as wrong?

### DISCUSSION THREAD

**Discussion by hohoho4u4me** [ANLAB-KAIST/KENSv3#218] [Project Q&A 2. Transport Layer (TCP)] [answered]
I have misunderstood the document and returned our 'syscall_write' after we got all the ack packets for the data packets we send. 
(ex -if we send 2 packets for write, return it after ack of those 2 come)
However this passes all the test.

Is this OK, or should we change the code to follow the document -> return write just after we stored data to send buffer?

**Comment by Pusnow** [comment, marked as answer]
The document describes one way to implement the write call. Your way is also correct.

As long as your implementation passes all cases, you don't need to change the code.

**Reply by hohoho4u4me** [reply]
Thank you for fast reply!!!!!
